zeebe/dynamic-config/src/main/java/io/camunda/zeebe/dynamic/config/RoutingStateInitializer.java

```diff
@@ -40,17 +46,30 @@ public RoutingStateInitializer(
 
   @Override
   public ActorFuture<ClusterConfiguration> modify(final ClusterConfiguration configuration) {
-    if (configuration.routingState().isPresent()) {
-      return CompletableActorFuture.completed(configuration);
-    }
-    // read it from the file
     RoutingState routingState = null;
+    AwaitRedistributionCompletion operation = null;
     if (routingStatePath.isPresent() && Files.exists(routingStatePath.get())) {
       final byte[] routingStateContent;
       try {
         routingStateContent = Files.readAllBytes(routingStatePath.get());
         routingState =
             serializer.deserializeRoutingState(routingStateContent, 0, routingStateContent.length);
+        operation =
+            switch (routingState.requestHandling()) {
+              case final RequestHandling.ActivePartitions activePartitions ->
+                  new AwaitRedistributionCompletion(
```

Comment:
This is problematic if the snapshot was taken before the scale up started. 
I think we should add these operations to the pending changes if, when replying the log, we encounter the scaling operations.